### PR TITLE
added xonsh recipe

### PR DESCRIPTION
**Pull request by scopatz**

### CODE REVIEW

**Comment by pelson** [comment]
Strange that it can't find setup.py on Windows. I wonder if you have to use a ``fn`` entry for the source.
Also, I don't think the bld.bat is needed in this case (it would come from script anyway). Last but not least, would you mind adding an import test and a ``commands`` test?

**Comment by scopatz** [comment]
Updated!

**Comment by scopatz** [comment]
Oooh so how do we disable legacy Python for this recipe?

**Comment by pelson** [comment]
> Oooh so how do we disable legacy Python for this recipe?

There are two ways, either use the skip section, or define a condition on build/run. The latter doesn't work very well though with conda-build at the moment (one to be improved). The skip section would look something like:

```
build:
   skip: True  # [py2k]
```

**Comment by scopatz** [comment]
Disabled Python 2

File: recipes/xonsh/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - ply
+
+test:
+  requires:
+    - nose
+  imports:
+    - xonsh
+  commands:
+    - xonsh -h
+  nosetests -w tests
```
Comment (pelson) [inline recipes/xonsh/meta.yaml]: I suspect this is a problem.

```
Traceback (most recent call last): 
  File "C:\conda\scripts\obvci_conda_build_dir.py", line 15, in <module>
    bld_dir.main()
  File "C:\conda\lib\site-packages\obvci\cli\conda_build_dir.py", line 19, in main
    return Builder.handle_args(args).main()
  File "C:\conda\lib\site-packages\obvci\conda_tools\build_directory.py", line 223, in main
    recipe_metas = self.fetch_all_metas()
  File "C:\conda\lib\site-packages\obvci\conda_tools\build_directory.py", line 196, in fetch_all_metas
    recipe_metas = fetch_metas(conda_recipes_root)
  File "C:\conda\lib\site-packages\obvci\conda_tools\build_directory.py", line 65, in fetch_metas
    packages.append(MetaData(package_dir))
  File "C:\conda\lib\site-packages\conda_build\metadata.py", line 349, in __init__
    self.parse_again()
  File "C:\conda\lib\site-packages\conda_build\metadata.py", line 357, in parse_again
    self.meta = parse(get_contents(self.meta_path))
  File "C:\conda\lib\site-packages\conda_build\metadata.py", line 133, in parse
    res = yamlize(data)
  File "C:\conda\lib\site-packages\conda\utils.py", line 119, in __call__
    value = self.func(*args, **kw)
  File "C:\conda\lib\site-packages\conda_build\metadata.py", line 96, in yamlize
    return yaml.load(data, Loader=BaseLoader)
  File "C:\conda\lib\site-packages\yaml\__init__.py", line 71, in load
    return loader.get_single_data()
  File "C:\conda\lib\site-packages\yaml\constructor.py", line 37, in get_single_data
    node = self.get_single_node()
  File "_yaml.pyx", line 707, in _yaml.CParser.get_single_node (ext/_yaml.c:8308)
  File "_yaml.pyx", line 725, in _yaml.CParser._compose_document (ext/_yaml.c:8581)
  File "_yaml.pyx", line 776, in _yaml.CParser._compose_node (ext/_yaml.c:9306)
  File "_yaml.pyx", line 890, in _yaml.CParser._compose_mapping_node (ext/_yaml.c:10838) 
  File "_yaml.pyx", line 776, in _yaml.CParser._compose_node (ext/_yaml.c:9306)
  File "_yaml.pyx", line 890, in _yaml.CParser._compose_mapping_node (ext/_yaml.c:10838) 
  File "_yaml.pyx", line 774, in _yaml.CParser._compose_node (ext/_yaml.c:9284)
  File "_yaml.pyx", line 853, in _yaml.CParser._compose_sequence_node (ext/_yaml.c:10333)
  File "_yaml.pyx", line 905, in _yaml.CParser._parse_next_event (ext/_yaml.c:11045) 
yaml.scanner.ScannerError: while scanning a simple key
  in "<unicode string>", line 29, column 3
could not find expected ':'
  in "<unicode string>", line 31, column 1
```

**Comment by pelson** [comment]
Looks like this isn't building on OSX and Linux. Very strange - I suspect we need to update some of our dependencies... would you mind rebasing this on upstream/master to see your base was out of date?

**Comment by scopatz** [comment]
OK.  I rebased and nothing changed. Not sure why this is a problem on windows but not elsewhere. I'll just remove it for now.

**Comment by scopatz** [comment]
Now seeing this gem which doesn't make sense to me since it is suppossed to be on Python 3.4

```console
conda install -c http://conda.binstar.org/pelson/channel/development --yes --quiet obvious-ci
Fetching package metadata: ...... 
Solving package specifications: ........... 
Error: Unsatisfiable package specifications.
Generating hint: 
[                    ]|                                                  |   0%
[1/3                 ]|################                                  |  33%
[      COMPLETE      ]|##################################################| 100%
 
 
 
Hint: the following packages conflict with each other:
  - obvious-ci
  - python 3.5*
 
Use 'conda info obvious-ci' etc. to see the dependencies for each package.
 
Note that the following features are enabled:
  - vc14
 
Command exited with code 1
```

**Comment by ocefpaf** [comment]
I just got affected by that too in [another recipe](https://ci.appveyor.com/project/comtbot/conda-recipes/build/1.0.1292).  Maybe [this](https://github.com/ioos/conda-recipes/pull/553#issuecomment-153535944) comment can help.

**Comment by scopatz** [comment]
@ocefpaf any ideas on how to actually fix this though?

**Comment by ocefpaf** [comment]
Not yet...  I am struggling with some hiccups in my day job.

**Comment by scopatz** [comment]
Good luck

**Comment by pelson** [comment]
I've made some changes in #16 which (combined with https://github.com/conda-forge/obvious-ci-feedstock) I believe will allow this to build successfully. Either way I'm happy to merge this and fix any resulting problems (even if that means disabling a subset of the build matrix for a period). Progress. :wink:

**Pull request merged**